ui/src/components/ConfigurationForm/ConfigurationForm.component.tsx

```diff
@@ -119,13 +165,53 @@ export const ConfigurationForm: React.FC<ConfigurationFormProps> = ({ isEmbedded
     });
   };
 
+  const [snackbar, setSnackbar] = useState<SnackbarConfig>({
+    open: false,
+    severity: 'info',
+  });
+  const handleSnackClose = (event?: SyntheticEvent | Event, reason?: string) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setSnackbar({ open: false });
+  };
+
+  const handleShareConfiguration = async () => {
+    const casted = form.getValues() as StarterRequest;
+    const urlToShare = stringifyUrl(
+      { url: window.location.href, query: { ...casted } },
+      { skipNull: true, skipEmptyString: true }
+    );
+    await navigator.clipboard.writeText(urlToShare);
+    setSnackbar({ open: true, severity: 'info', message: 'Link to configuration copied to clipboard.' });
+  };
+
   return (
     <Box>
       {!isEmbedded && (
-        <Typography variant="h3" component="h3" fontWeight={300} gutterBottom>
-          Generate tapir project
-        </Typography>
+        <Stack direction="row" spacing={0.25}>
```

Comment:
Have you considered extracting this tag and its contents and functionality to separate component?

Reply:
I was rather thinking about componentizing `Tooltip` but I would like to do that as a follow up rather than here

Reply:
@tpacuszkapl we have discussed with @kkolasinski that componentization works nice with the other parts of that form (drop downs, or radio groups, etc...) as they are enclose independent parts and they are re-used. However neither header nor `share-button` are subject of the same as they are not going to be re-used in the near future (there are no plans to have more forms here) and the `share-button` heavily depends on the form content so it kind of makes sense to keep it there... IOW unless you have another angle to view on it (that we are missing) I would refrain from componentizing it...

Reply:
Fine with me